Assets/Tests/InputSystem/CoreTests_GlobalActions.cs

```diff
@@ -0,0 +1,500 @@
+    public void GlobalActions_CanAddNewInteraction_ToAllBindingsOnActionAtOnce()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        var action = new InputAction();
+        action.AddBinding("<keyboard>/a");
+        action.AddCompositeBinding("1DAxis")
+            .With("negative", "<keyboard>/w")
+            .With("positive", "<keyboard>/s");
+        action.AddBinding("<keyboard>/b");
+
+
+        var input = new Input<float>(action);
+        input.AddInteraction<HoldInteraction>();
+
+        var holdPerformed = false;
+        input.action.performed += ctx =>
+        {
+            holdPerformed = true;
+        };
+
+        Press(keyboard.aKey, currentTime);
+        currentTime += 0.5; // default hold time is 0.4
+        InputSystem.Update();
+
+        Assert.That(holdPerformed, Is.True);
+
+        Release(keyboard.aKey);
+        holdPerformed = false;
+        Press(keyboard.wKey, currentTime);
+        currentTime += 0.5;
+        InputSystem.Update();
+
+        Assert.That(holdPerformed, Is.True);
+
+        Release(keyboard.wKey);
+        holdPerformed = false;
+        Press(keyboard.bKey, currentTime);
+        currentTime += 0.5;
+        InputSystem.Update();
+
+        Assert.That(holdPerformed, Is.True);
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_CanAddNewInteraction_ToSpecificBindingsOnAction()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        var action = new InputAction();
+        action.AddBinding("<keyboard>/a");
+        action.AddCompositeBinding("1DAxis")
+            .With("negative", "<keyboard>/w")
+            .With("positive", "<keyboard>/s");
+        action.AddBinding("<keyboard>/b");
+
+
+        var input = new Input<float>(action);
+        input.AddInteraction<HoldInteraction>(input.bindings[1]);
+
+        var holdPerformed = false;
+        input.action.performed += ctx =>
+        {
+            // when this callback fires from a binding that didn't have the interaction added to it, the
+            // CallbackContext interaction should be null
+            holdPerformed = ctx.interaction != null;
+        };
+
+        Press(keyboard.aKey, currentTime);
+        currentTime += 0.5; // default hold time is 0.4
+        InputSystem.Update();
+
+        // make sure this didn't fire for the a key
+        Assert.That(holdPerformed, Is.False);
+
+        Release(keyboard.aKey);
+        holdPerformed = false;
+        Press(keyboard.wKey, currentTime);
+        currentTime += 0.5;
+        InputSystem.Update();
+
+        Assert.That(holdPerformed, Is.True);
+
+        Release(keyboard.wKey);
+        holdPerformed = false;
+        Press(keyboard.bKey, currentTime);
+        currentTime += 0.5;
+        InputSystem.Update();
+
+        Assert.That(holdPerformed, Is.False);
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_CanPollInteractionPhaseChanges()
```

Comment:
I wonder if we should split 'interaction' tests into a separate file. The test themselves seem reasonable